packages/js/product-editor/src/components/header/header.tsx

```diff
@@ -17,11 +19,23 @@ import { PublishButton } from './publish-button';
 import { Tabs } from '../tabs';
 
 export type HeaderProps = {
+	productId: Product[ 'id' ];
 	onTabSelect: ( tabId: string | null ) => void;
-	productName: string;
 };
 
-export function Header( { onTabSelect, productName }: HeaderProps ) {
+export function Header( { productId, onTabSelect }: HeaderProps ) {
```

Comment:
Just wanted to reduce side effects as much as possible. Also the productId from props is being stored in a different state `useProductEntityRecord` in comparison to the id coming from `useEntityProp` so yes they can be different because we don't have a single source of truth. This is the reason why previously we use `productName` from the props instead of getting it from the `useEntityProp`.

Reply:
>Also the productId from props is being stored in a different state useProductEntityRecord in comparison to the id coming from useEntityProp so yes they can be different because we don't have a single source of truth

This is a good point about the single source of truth (though we'd probably have a pretty big issue if they did not match).

I'd prefer to keep these components context aware like other block components and refer to the block entity data store instead of the props passed in.  Currently, these are the only components making used of passed props and it seems to be convention to rely on the entity data store for any components within the entity provider.